File: platforms/web/examples/targets/texture_target/texture.js
import { Renderer, Shader } from "fragmentcolor";

const renderer = new Renderer();

// Render some scene into an offscreen target, then read it back as a
// texture in a second fullscreen pass.
const offscreen = await renderer.createTextureTarget([512, 512]);
const output = await renderer.createTextureTarget([512, 512]);

const postShader = new Shader(`
struct VOut {
    @builtin(position) pos: vec4<f32>,
    @location(0) uv: vec2<f32>,
};

@vertex
fn vs_main(@builtin(vertex_index) i: u32) -> VOut {
    var p  = array<vec2<f32>, 3>(vec2<f32>(-1., -1.), vec2<f32>(3., -1.), vec2<f32>(-1., 3.));
    var uv = array<vec2<f32>, 3>(vec2<f32>(0., 1.), vec2<f32>(2., 1.), vec2<f32>(0., -1.));
    var out: VOut;
    out.pos = vec4<f32>(p[i], 0., 1.);
    out.uv = uv[i];
    return out;
}

@group(0) @binding(0) var t: texture_2d<f32>;
@group(0) @binding(1) var samp: sampler;

@fragment
fn fs_main(in: VOut) -> @location(0) vec4<f32> {
    return textureSample(t, samp, in.uv);
}
`);

// `texture()` hands back the offscreen target's color texture so it can be
// bound as a shader uniform on web.
const tex = offscreen.texture();
await postShader.set("t", tex);
renderer.render(postShader, output);

File: platforms/swift/examples/targets/texture_target/texture.swift
import FragmentColor

let renderer = Renderer()
let target = try await renderer.createTextureTarget([256, 256])

// Bind the offscreen target's contents as a uniform on a downstream
// post-processing shader.
let post = try Shader("""
    @group(0) @binding(0) var input_image : texture_2d<f32>
    @group(0) @binding(1) var input_sampler : sampler

    @vertex fn vs_main(@builtin(vertex_index) i: u32) -> @builtin(position) vec4<f32> {
        let p = array<vec2<f32>, 3>(vec2f(-1.0,-1.0), vec2f(3.0,-1.0), vec2f(-1.0,3.0))
        return vec4<f32>(p[i], 0.0, 1.0)
    }
    @fragment fn fs_main(@builtin(position) pos: vec4<f32>) -> @location(0) vec4<f32> {
        return textureSample(input_image, input_sampler, vec2<f32>(0.5, 0.5))
    }

""")
try post.set("input_image", target.texture())